src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/EnumConverterTests.cs

```diff
@@ -635,5 +652,18 @@ private class ZeroAppenderPolicy : JsonNamingPolicy
         {
             public override string ConvertName(string name) => name + "0";
         }
+
+        private static JsonSerializerOptions CreateStringEnumOptionsForType<TEnum>(bool useGenericVariant, JsonNamingPolicy? namingPolicy = null, bool allowIntegerValues = true) where TEnum : struct, Enum
+        {
+            return new JsonSerializerOptions
+            {
+                Converters =
+                {
+                    useGenericVariant
+                    ? new JsonStringEnumConverter<TEnum>(namingPolicy, allowIntegerValues)
+                    : new JsonStringEnumConverter(namingPolicy, allowIntegerValues)
+                }
+            };
+        }
```

Comment:
To validate that the new converter works in AOT you mean? That should be covered by the main test suite.

Reply:
We'd typically write a trimming test if there's a warning suppression to ensure the suppression is valid - purely because in the huge xunit test there could be code that keeps things we'd have otherwise thrown away. Trimming tests have very little code in it and that makes it more likely things will get trimmed and test would fail if the suppression was invalid. I don't see a suppression here so that xunit test should be sufficient.